test/psych/test_hash.rb

```diff
@@ -111,5 +111,19 @@ def test_ref_append
 eoyml
       assert_equal({"foo"=>{"hello"=>"world"}, "bar"=>{"hello"=>"world"}}, hash)
     end
+
+    def test_key_deduplication
+      unless String.method_defined?(:-@) && (-("a" * 20)).equal?((-("a" * 20)))
+        skip "This Ruby implementation doesn't support string deduplication"
+      end
```

Comment:
Do we need this test in the implementation too?  IOW, will the `to_ruby` visitor break if `:-@` isn't defined?

Reply:
Hum, somehow I was certain I did put it there, but seems like not. I'll update in a sec.